ihaskell.cabal

```diff
@@ -127,7 +127,7 @@ executable ihaskell
   hs-source-dirs: main
   other-modules:
                    Paths_ihaskell
-  ghc-options: -threaded -rtsopts -Wall -dynamic
+  ghc-options: -threaded -rtsopts -Wall
```

Comment:
Why was this flag removed?

Reply:
IMO this flag shouldn't have been added in the first place. If someone needs this flag, they can add it with `--ghc-options -dynamic`.

It simplifies the Docker image by building a statically linked binary, so that I don't have to copy all the `.so` files into the runtime image. There's probably a `-no-dynamic` flag that I could use to disable it, but, again, executables shouldn't be built dynamically by default

Reply:
I added it for a reason: https://gitlab.haskell.org/ghc/ghc/-/issues/15524

Reply:
hm the PR that added this flag doesn't mention that issue at all: https://github.com/gibiansky/IHaskell/pull/1184

Seems to only affect the Nix build, is that right? I notice the Nix builds in this CI increased from ~1min to ~3min, but the Linux build takes roughly the same amount of time

Reply:
Did you read the linked issue? The problem isn't Nix-specific.

Reply:
I did read the issue, but I said

> I notice the Nix builds in this CI increased from ~1min to ~3min, but the Linux build takes roughly the same amount of time.

So I don't see any performance regression in the non-Nix side. I also don't see any obvious performance issues using the Docker image built from this branch

Reply:
Okay, I tried it myself without `-dynamic` and the performance on Nix was fine, I didn't notice any issues either.

Reply:
@brandon-leapyear @vaibhavsagar 
Hi, in my use case like the following link, when the `-dynamic` is deleted, ihaskell does not load some packages.
https://github.com/hasktorch/hasktorch-jupyter/blob/master/Dockerfile

ihaskell needs to load packages at runtime and does not work entirely with a single binary.
It certainly works with ghc-options -dynamic, but I had to rebuild again this time.

Reply:
I think anything using FFI is affected as there is a problem loading c/c++ DLLs.

Reply:
This option is needed to link c/c++ system libraries not haskell libraries.

Reply:
What do you mean by "I had to rebuild again this time"? It seems like you're [installing `ihaskell` from source](https://github.com/hasktorch/hasktorch-jupyter/blob/42dfb4ab7c213a3b91f2a105126d39857d51ff47/Dockerfile#L59-L67) already, so it's not like you're rebuilding an artifact that was already built by this Docker image.

If you mean that you built the Docker image once, realized it didn't work, and then have to rebuild the Docker image with `--ghc-options -dynamic`, that's just normal Docker development. If you're writing a `Dockerfile` and you used the wrong arguments in a command, you just have to go back and rebuild the image with the correct arguments.

Unless, are you saying that `ihaskell` depends on something that needs to be loaded dynamically? I haven't been running into this, it must be specific to your project. If it's something specific to your project, you should add `--ghc-options -dynamic` yourself; you shouldn't force everyone building `ihaskell` to use `-dynamic` just because you need it

Reply:
In other words, without this option, ihaskell will not be able to link to libraries that ihaskell does not directly link.
When ihaskell is built, it doesn't know what haskell packages are loaded, and it don't check what c/c++ libraries those haskell packages link to.